### PR TITLE
Add github checks output support to stability check

### PR DESCRIPTION
**Pull request by stephenmcgruer**

### CODE REVIEW

**Comment by B4LQ15** [comment]
Good job

File: tools/ci/tc/tasks/test.yml [hunk 1]
```diff
@@ -10,6 +10,13 @@ components:
       public/results:
         path: /home/test/artifacts
         type: directory
+    extra:
+      github:
+        customCheckRun:
+          # We set both textArtifactName and annotationsArtifactName due
+          # to https://github.com/taskcluster/taskcluster/issues/3191
```
Comment (Hexcles) [inline tools/ci/tc/tasks/test.yml:17]: Drive-by: that issue seems to have been fixed?

File: dom/historical.html [hunk 2]
```diff
@@ -64,7 +64,11 @@
 
 test(function() {
   // https://github.com/whatwg/html/commit/e236f46820b93d6fe2e2caae0363331075c6c4fb
-  assert_false("load" in document);
+  if (Math.random() > 0.5) {
```
Comment (jgraham) [inline dom/historical.html:67]: This needs to be reverted ofc.

File: tools/wptrunner/wptrunner/wptlogging.py [hunk 3]
```diff
@@ -65,3 +65,18 @@ def __call__(self, data):
             not self.has_log and
             log_levels[data["level"]] <= self.min_level):
             self.has_log = True
+
+
+__tc_logger = None
+def get_taskcluster_logger(kwargs):
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/wptlogging.py:71]: Using the logging library here seems pretty sketchy, since in theory we're trying to catch logging output are redirect it to mozlog. I think for this case I'd rather just have a custom object to implement writing to the taskcluster markdown file; using logging doesn't really solve much since we don't depend on any of the features of logging.

**Review by jgraham — CHANGES_REQUESTED** [review]

**Review by stephenmcgruer — COMMENTED** [review]
Please take another look. I don't love my code here...

On a more meta level, I'm also having trouble in framing the instructions for test authors. Specifically, I want to be able to say something like "please use the printed WPT command to reproduce", but the command as dumped will have multiple flags they probably don't want to use (i.e. `--github-checks-text-file=/home/test/artifacts/checkrun.md` and `--affected base_head`). So far I've changed it to a weaker sentence, but maybe I should be even more explicit, something like:

> Please try to reproduce; see the above WPT command for what the bot ran, but note you may need to remove some flags to run locally.

Which is still confusing to test authors :/.

**Review by jgraham — APPROVED** [review]
This is pretty exciting!

I don't have a great suggestion for showing the command that you should run to get things working locally without some terrible hack where you have a blocklist of flags to remove. We never quite managed to solve this issue for gecko ci (although in that case things are worse because the ci uses a totally different entrypoint compared to the local runs).

**Comment by stephenmcgruer** [comment]
Reverted the test change, will land this once CI is green. For posterity, this is what it looks like:

image
